File: test-client/src/models/viewModels/MessageVM.ts
import { IMessage } from "models/interfaces/IMessage";

export interface MessageVM extends IMessage {
    _id: string, 
    type: string,
    message: string,
    timestamp: string, 
    filename?: string,
    url?: string,    
    
    sender: string,   
    room: string, 
    
    delivered: boolean
}

File: websocket/model/view_model/MessageVM.py
from dataclasses import dataclass, field
from typing import Optional

@dataclass
class MessageVM():
    _id: str 
    type: str
    message: str
    timestamp: str

    sender: str    # User ID
    room: str      # Room ID

    filename: Optional[str] = field(default=None)
    url: Optional[str] = field(default=None)